### DISCUSSION TITLE
Decorate the annotation

### DISCUSSION THREAD

**Discussion by ghost** [jokergoo/ComplexHeatmap#1004] [Q&A]
![image](https://user-images.githubusercontent.com/56685916/199470079-4fcabcc5-6657-4da7-b12e-0937d1cc5f0f.png)

Hi all,

I am able to use the complexheatmap to create such a heatmap. What I want to do is to make the annotation on the top filled with white color and grid border to be black.

Here is my code,

```r
column_ann = HeatmapAnnotation(type2 = oncoset_1$Category_2,type1 = oncoset_1$Category_1,simple_anno_size = unit(1, "cm"),
                               border = T,show_legend = c(FALSE, FALSE),gap = unit(0, "mm"),show_annotation_name = F)

Heatmap(heatmapdata_1,
        width = ncol(heatmapdata_1)*unit(4.5, "mm"), 
        height = nrow(heatmapdata_1)*unit(4.5, "mm"),
        rect_gp = gpar(col = "black", lwd = 1),
        row_split = rep(c("Ctrl", "Parental","Secondary"), each=2),
        row_gap = unit(2, "mm"), 
        column_order = colnames(heatmapdata_1),
        row_order =rownames(heatmapdata_1),
        col = brewer.pal(n = 8, name = "RdYlBu"),
        heatmap_legend_param = list(
          title = "Cell viabibilty",
          legend_height = unit(3, "cm")),
        border = TRUE,
        show_row_names = F,
        row_title_rot = 0,
        top_annotation = column_ann)
```

I expect it to be like

![image](https://user-images.githubusercontent.com/56685916/199471138-aeebd468-c5ce-48a6-a2b5-9a7a7a1c2ad7.png)

Those text and grid I draw in photoshop. But it would be very nice if I can output like this with complexheatmap. Doesn't have to be exactly same.

Thanks in advance!!!

**Comment by jokergoo** [comment]
This requirement is very very specific that I will not implement it in the package. As you said, these are actually grids and texts, you can do it by:

1. create a empty annotation to just allocate an empty plotting region there:


```r
Heatmap(m, top_annotation = HeatmapAnnotation(empty = anno_empty()))
```

2. Later use `decorate_annotation()` to draw grids and texts there:

```r
decorate_annotatio("empty", {
    grid.rect(...)
    grid.text(...)
    ...
})
```
